pkg/authentication/scram/scram_credentials.go

```diff
@@ -0,0 +1,196 @@
+package scram
```

Comment:
(almost) all of this was extracted from enterprise. We can import it from there, I will highlight the bits that are different